__tests__/Auth0Client.test.ts

```diff
@@ -173,28 +173,389 @@ describe('Auth0Client', () => {
+
+      expect(tokenVerifier).toHaveBeenCalledWith(
+        expect.objectContaining({
+          max_age: undefined
+        })
+      );
+    });
+    it('calls `tokenVerifier.verify` with the parsed `max_age` string from constructor', async () => {
+      const auth0 = setup({ max_age: '10' });
+
+      await loginWithPopup(auth0);
+
+      expect(tokenVerifier).toHaveBeenCalledWith(
+        expect.objectContaining({
+          max_age: 10
+        })
+      );
+    });
+    it('calls `tokenVerifier.verify` with the parsed `max_age` number from constructor', async () => {
+      const auth0 = setup({ max_age: 10 });
+
+      await loginWithPopup(auth0);
+
+      expect(tokenVerifier).toHaveBeenCalledWith(
+        expect.objectContaining({
+          max_age: 10
+        })
+      );
+    });
+
+    it('saves into cache', async () => {
+      const auth0 = setup();
+
+      jest.spyOn(auth0['cache'], 'save');
+
+      await loginWithPopup(auth0);
+
+      expect(auth0['cache']['save']).toHaveBeenCalledWith(
+        expect.objectContaining({
+          client_id: TEST_CLIENT_ID,
+          access_token: TEST_ACCESS_TOKEN,
+          expires_in: 86400,
+          audience: 'default',
+          id_token: TEST_ID_TOKEN,
+          scope: TEST_SCOPES
+        })
+      );
+    });
+
+    it('saves decoded token into cache', async () => {
+      const auth0 = setup();
+
+      const mockDecodedToken = {
+        claims: { sub: 'sub', aud: 'aus' },
+        user: { sub: 'sub' }
+      };
+      tokenVerifier.mockReturnValue(mockDecodedToken);
+
+      jest.spyOn(auth0['cache'], 'save');
+
+      await loginWithPopup(auth0);
+
+      expect(auth0['cache']['save']).toHaveBeenCalledWith(
+        expect.objectContaining({
+          decodedToken: mockDecodedToken
+        })
+      );
+    });
+
+    it('should not save refresh_token in memory cache', async () => {
+      const auth0 = setup({
+        useRefreshTokens: true
+      });
+
+      jest.spyOn(auth0['cache'], 'save');
+
+      await loginWithPopup(auth0);
+
+      expect(auth0['cache']['save']).toHaveBeenCalled();
+      expect(auth0['cache']['save']).not.toHaveBeenCalledWith(
+        expect.objectContaining({
+          refresh_token: TEST_REFRESH_TOKEN
+        })
+      );
+    });
+    it('should save refresh_token in local storage cache', async () => {
+      const auth0 = setup({
+        useRefreshTokens: true,
+        cacheLocation: 'localstorage'
+      });
+
+      jest.spyOn(auth0['cache'], 'save');
+
+      await loginWithPopup(auth0);
+
+      expect(auth0['cache']['save']).toHaveBeenCalledWith(
+        expect.objectContaining({
+          refresh_token: TEST_REFRESH_TOKEN
+        })
+      );
+    });
+
+    it('saves `auth0.is.authenticated` key in storage', async () => {
+      const auth0 = setup();
+
+      jest.spyOn(auth0['cookieStorage'], 'save');
+
+      await loginWithPopup(auth0);
+
+      expect(auth0['cookieStorage']['save']).toHaveBeenCalledWith(
```

Comment:
Same here, `cookieStorage` is private.

Reply:
As opposed to the previous case, this something we could check as we already mock `es-cookie` - we could check that the `set` method was called with the right data. It's a little awkward as it relies on checking a serialized piece of JSON, but there are things we can do to make that less brittle.

Reply:
I have updated this, however now this test is dependent on the implementation detail of using `CookieStorage` or `CookieStorageWithLegacySameSite`. So to that extend, I would argue it makes more sense to stick with mocking `Auth0Client.cookieStorage`.